mongodump/mongodump_test.go

```diff
@@ -1400,3 +1401,51 @@ func TestMongoDumpCollectionOutputPath(t *testing.T) {
 		})
 	})
 }
+
+func TestMongoDumpAwsAuth(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.AWSAuthTestType)
+	Convey("Should be able to run mongodump with AWS STS AssumeRole auth", t, func() {
```

Comment:
Imo, you don't need to create extra test cases for AWS Auth since the change only applies during connection. You can change `simpleMongoDumpInstance` to make a special case for opts if a special `AWS_AUTH` flag is set just like what `GetSSLOptions` did. In such way, you don't need `testtype.AWSAuthTestType` either, you can just run any integration test in the AWS_AUTH mode.
Same applies to other tests.

Reply:
Hmm, I was following the example set by the Kerberos tests -- any reason why there's a separate test for that kind of connection, as opposed to just setting a flag? I figured we didn't really need to run every integration test against a mongod in AWS mode, since we are just testing the connection, as you said, and a simple test for each tool would suffice. I think keeping it this way would simplify the structure of testing; toggling a flag might be less convenient. Thoughts?

Reply:
I got your reason for creating a new test case for the AWS auth now. Not sure about the reason for Kerberos to implement its own test case, but from what I see, reusing the existing test cases and added a flag for AWS Auth specific options will definitely simplifying the test structures. You might not need `run-mongodb-aws-test.sh`at all (with some tweak in the runTest.sh) and you don't need to create extra test cases for each Tools product. 
But definitely current implementation also works. 
I'd like to hear @tfogo 's thought on this matter.

Reply:
As discussed, this would be a separate ticket.